tests/components/hunterdouglas_powerview/test_config_flow.py

```diff
@@ -0,0 +1,220 @@
+        "host": "1.2.3.4",
+    }
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_homekit(hass):
+    """Test we get the form with homekit source."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    mock_powerview_userdata = _get_mock_powerview_userdata()
+    with patch(
+        "homeassistant.components.hunterdouglas_powerview.UserData",
+        return_value=mock_powerview_userdata,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": "homekit"},
+            data={
+                "host": "1.2.3.4",
+                "properties": {"id": "AA::BB::CC::DD::EE::FF"},
+                "name": "PowerViewHub._hap._tcp.local.",
+            },
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "link"
+    assert result["errors"] is None
+    assert result["description_placeholders"] == {
+        "host": "1.2.3.4",
+        "name": "PowerViewHub",
+    }
+
+    with patch(
+        "homeassistant.components.hunterdouglas_powerview.UserData",
+        return_value=mock_powerview_userdata,
+    ), patch(
+        "homeassistant.components.hunterdouglas_powerview.async_setup",
+        return_value=True,
+    ) as mock_setup, patch(
+        "homeassistant.components.hunterdouglas_powerview.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "PowerViewHub"
+    assert result2["data"] == {"host": "1.2.3.4"}
+    assert result2["result"].unique_id == "ABC123"
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+    result3 = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": "homekit"},
+        data={
+            "host": "1.2.3.4",
+            "properties": {"id": "AA::BB::CC::DD::EE::FF"},
+            "name": "PowerViewHub._hap._tcp.local.",
+        },
+    )
+    assert result3["type"] == "abort"
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    mock_powerview_userdata = _get_mock_powerview_userdata(
+        get_resources=asyncio.TimeoutError
+    )
+    with patch(
+        "homeassistant.components.hunterdouglas_powerview.UserData",
+        return_value=mock_powerview_userdata,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": "1.2.3.4"},
+        )
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5444210)